plugins/libplugin-pay.c

```diff
@@ -2671,16 +2672,61 @@ static struct presplit_mod_data *presplit_mod_data_init(struct payment *p)
 
 static u32 payment_max_htlcs(const struct payment *p)
 {
+	const struct payment *root;
```

Comment:
```suggestion
	const struct payment *root = payment_root(p);
```

Reply:
`payment_root` accepts and returns non-`const` pointers to `struct payment`, which is why I open-coded this here.

Since `payment_max_htlcs` does not modify the payment it makes sense for it to accept a non-`const` pointer.  I notice that I am probably the only one who consistently considers `const`-correctness.

As currently only `payment_max_htlcs` uses `const struct payment *`, it seems a bit wasteful to factor out a new `cpayment_root` that accepts and returns `const struct payment *`.

Reply:
> `payment_root` accepts and returns non-`const` pointers to `struct payment`, which is why I open-coded this here.

`payment_root` should accept non-`const`, that's my bad.

> it seems a bit wasteful to factor out a new `cpayment_root` that accepts and returns `const struct payment *`.

Definitely don't do this. I lieu of a better solution we can keep it as it is.